apps/web/src/pages/Pools/PoolAdvertisementPage/components/ApplicationLink.tsx

```diff
@@ -0,0 +1,66 @@
+import React from "react";
+import { useIntl } from "react-intl";
+
+import { Link } from "@gc-digital-talent/ui";
+
+import useRoutes from "~/hooks/useRoutes";
+import { Scalars } from "~/api/generated";
+import { useFeatureFlags } from "@gc-digital-talent/env";
+
+interface ApplicationLinkProps {
+  poolId: Scalars["ID"];
+  applicationId?: Scalars["ID"];
+  hasApplied?: boolean;
+  canApply?: boolean;
+}
+
+const ApplicationLink = ({
+  poolId,
+  applicationId,
+  hasApplied,
+  canApply,
+}: ApplicationLinkProps) => {
+  const intl = useIntl();
+  const paths = useRoutes();
+  const { applicationRevamp } = useFeatureFlags();
```

Comment:
Outside scope of this issue, but might be nice if useFeatureFlags uses context, and then it could be forced in stories without messing around with window global state, which is glitchy.

Reply:
Would this overlap with https://github.com/gctc-ntgc/gc-digital-talent/issues/6387 ?

Reply:
They'd definitely conflict if we were working on both at the same time, but it does seem like two different tickets to me